packages/ToggleSwitch/ToggleSwitch.jsx

```diff
@@ -1,131 +1,98 @@
-import React, { Component } from 'react'
+import React, { useState } from 'react'
 import PropTypes from 'prop-types'
 
 import { safeRest } from '@tds/util-helpers'
 import Text from '@tds/core-text'
 import Box from '@tds/core-box'
 import Tooltip from '@tds/core-tooltip'
 import Spinner from '@tds/core-spinner'
-import FlexGrid from '@tds/core-flex-grid'
-import { Button, Slider, Switch, InputSwitchWrapper, SpinnerWrapper } from './styles'
+import { StyledLabel, Button, Slider, InputSwitchWrapper } from './styles'
 
 /**
-  * ToggleSwitch is an alternative to using a checkbox, and maintains a similar component interface to [@tds/core-checkbox](https://tds.telus.com/components/index.html#checkbox).
-
-   This component will _only_ maintain internal state when an `onChange` handler is _not_ provided.
-   When an `onChange` handler is passed, it becomes the app's responsibility to manage this component's state through props.
-   This behaviour differs from `@tds/core-checkbox`.
-  * @version ./package.json
-  */
-class ToggleSwitch extends Component {
-  componentWillMount() {
-    const { checked } = this.props
-    this.setState({ checked })
+ * @version ./package.json
+ */
+
+const ToggleSwitch = ({
+  id,
+  label,
+  toolTipText,
+  checked,
+  onBlur,
+  onClick,
+  onFocus,
+  toolTipCopy,
+  spinnerLabel,
+  ...rest
+}) => {
+  const labelledById = `${id}-label`
+
+  const [isPressed, setIsPressed] = useState(checked)
+  const [isLoading, setIsLoading] = useState(false)
+  const [isSpinning, setIsSpinning] = useState(false)
+
+  React.useEffect(() => {
+    setIsSpinning(false)
+    setIsLoading(false)
+  }, [checked])
+
+  React.useEffect(() => {
+    const timer = setTimeout(() => {
+      if (isLoading) {
+        setIsSpinning(true)
+      } else {
+        setIsSpinning(false)
+      }
+    }, 250) // time needed for a slider to move
+    return () => clearTimeout(timer)
+  }, [isLoading])
+
+  const handleClick = event => {
+    setIsPressed(!isPressed)
+    setIsLoading(true)
+    onClick(event)
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.state.checked !== nextProps.checked) {
-      this.setState({ checked: nextProps.checked })
-    }
-  }
-
-  onChange = event => {
-    const { onChange } = this.props
-    if (onChange) {
-      event.persist()
-      onChange(event)
-    } else {
-      this.setState(({ checked }) => ({ checked: !checked }))
-    }
-  }
-
-  onFocus = event => {
-    const { onFocus } = this.props
-
+  const handleFocus = event => {
     if (onFocus) {
       event.persist()
       onFocus(event)
     }
   }
 
-  onBlur = event => {
-    const { onBlur } = this.props
-
+  const handleBlur = event => {
```

Comment:
Are these handleBlur and handleFocus needed?

Reply:
Not needed.